blips/blip-0001.md

```diff
@@ -0,0 +1,289 @@
+```
+bLIP: 1
+Title: bLIP process
+Status: Active
+Author: Ryan Gentry <ryan@lightning.engineering>
+Created: 2021-05-21
+Post-History: 2021-06-30: https://lists.linuxfoundation.org/pipermail/lightning-dev/2021-June/003086.html
+  [lightning-dev] bLIPs: A proposal for community-driven app layer and protocol extension standardization
+License: CC0
+```
+
+# Abstract
+
+bLIP stands for Bitcoin Lightning Improvement Proposal. A bLIP is a design document 
+providing information to the Lightning community, or describing a new feature for 
+the Lightning Network. The bLIP should provide a concise technical specification of 
+the feature and a rationale for the feature. The bLIP author is responsible for 
+building consensus within the community and documenting dissenting opinions.
+
+# Copyright
+
+This bLIP is licensed under the CC0 license.
+
+# Rationale
+
+As the Lightning community has grown, new features, standards, and protocols have 
+been developed outside of the BOLT specification process: particularly at the 
+application level that isn’t described within the core BOLT documents. This is great! 
+But in the spirit of interoperability, documenting features, standards, and protocols 
+in a single location with a standard format will make it easy on future adopters. 
+
+In particular, there are three scarce sets of identifiers used in Lightning Network
```

Comment:
There's a fourth one: BOLT 4 onion failure messages (https://github.com/lightningnetwork/lightning-rfc/blob/master/04-onion-routing.md#failure-messages)

Reply:
Good idea, these do make sense.
I'm more and more leaning towards having one bLIP to centralize all these assigned values.

Reply:
hmm... should application devs experimenting with BOLT 4 onion failure messages be writing bLIPs? Given that they're almost entirely used to inform pathfinding (IIUC), a proposal adding new onion failure messages strike me more as "a feature... intended to become universal or near universal" and should go through the existing BOLTs.

Happy to add `channel_update` flags and invoice's tagged fields, though I am concerned about scope creep with this initial doc... we can always add more to it if demand justifies it.

Reply:
True, maybe we should take a more iterative approach here, and start only with things that are obviously needed by bLIPs (the ones you've currently listed) and we can add more later when (if) a future bLIP actually needs it.

Reply:
How about this as a compromise: I'll add a sentence about how potentially more scarce identifiers exist (e.g. BOLT4 onion failure messages, channel_update flags, and invoice tagged fields), and if/when bLIPs need them we can further specify how and where to assign/allocate those identifiers.

Reply:
SGTM